.github/workflows/rust-sdk-tests.yml

```diff
@@ -0,0 +1,82 @@
+# Copyright (C) 2025 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This workflow is triggered by analyze.yml
+name: Perfetto Rust SDK CI
+on:
+  workflow_call:
+env:
+  NIGHTLY_VERSION: nightly-2025-01-03
+  NIGHTLY_FEATURES: "--features=intrinsics"
+jobs:
+  build-test:
+    name: Rust ${{ matrix.toolchain }} - ${{ matrix.name }}
+    runs-on: self-hosted
+    timeout-minutes: 20
+
+    strategy:
+      matrix:
+        include:
+          - name: stable
+            toolchain: stable
+            features: ""
+          - name: nightly
+            toolchain: $NIGHTLY_VERSION
+            features: $NIGHTLY_FEATURES
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v4
+
+      - name: Install Rust (${{ matrix.name }})
+        uses: dtolnay/rust-toolchain@master
```

Comment:
I would feel more comfortable if this was pinned given it's not a "mainstream" dependency to guard against supply chain attacks.